src/App.tsx

```diff
@@ -15,49 +23,82 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface IGoodsQuery {
+  sortBy: string;
+  isReversed: boolean;
+}
+
+const goods: IGoods[] = goodsFromServer.map(name => (
+  {
+    key: getRandomKey(),
+    name,
+  }
+));
+
+function getPreparedGoods(
+  goodsToPrepared: IGoods[],
+  {
+    sortBy,
+    isReversed,
+  }: IGoodsQuery,
+) {
+  const sortedGoods = [...goodsToPrepared];
+
+  if (sortBy) {
+    sortedGoods.sort((goodsA, goodsB) => {
+      switch (sortBy) {
+        case EButtonsSortBy.Alphabet:
+          return goodsA.name.localeCompare(goodsB.name);
+
+        case EButtonsSortBy.Length:
+          return goodsA.name.length - goodsB.name.length;
+
+        default: throw Error('Smth gone wrong');
```

Comment:
It is better to give a more informative error text (incorrect sort type received, etc.)